include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -518,8 +518,14 @@ struct __usm_or_buffer_accessor
     }
 };
 
+struct __result_and_scratch_storage_base
```

Comment:
Why do we introduce that class with virtual destructor?
On first glance I cannot see a polymorphic behavior... 
Even it there is, I would like to warn  - "virtual functions" don't work within a kernel..

Reply:
`virtual` required here only on host side.